### PR TITLE
scripts: Optionally disallow additionalProperties at the root

### PR DESCRIPTION
**Pull request by maxbrunet**
It can useful to add `additionalProperties` to the root of the schema to detect extraneous keys.

(the new tests are very nice!)

This also adds a new line at the end of the schema files.

### CODE REVIEW

**Comment by maxbrunet** [comment]
I have made the change to be opt-in only as it is not absolutely required for a CRD to have a complete schema for build-in fields, since these are implicitly validated. See strimzi/strimzi-kafka-operator#5997 for example. Although, most have them.

**Comment by yannh** [comment]
Hi Max, thanks! Could you add a couple of tests to demonstrate the new behaviour of additional_properties?  :pray: Thanks :bow:

File: scripts/fixtures/prometheus_v1-denyRootAdditionalProperties.json [hunk 1]
```diff
@@ -0,0 +1,7098 @@
...
+            "required": [
+              "url"
+            ],
+            "type": "object",
+            "additionalProperties": false
+          },
+          "type": "array"
+        },
+        "replicaExternalLabelName": {
+          "description": "Name of Prometheus external label used to denote repl
```
Comment (maxbrunet) [inline scripts/fixtures/prometheus_v1-denyRootAdditionalProperties.json:7097]: And here it is!

**Comment by maxbrunet** [comment]
Done! I have added tests for both `FILENAME_FORMAT` and `DENY_ROOT_ADDITIONAL_PROPERTIES`

**Review by yannh — APPROVED** [review]

**Comment by yannh** [comment]
Sweet thanks! A unit test would be nice some day but this is good already :100:

**Pull request merged**